xds/internal/resolver/serviceconfig.go

```diff
@@ -381,17 +371,19 @@ func (r *xdsResolver) newConfigSelector(su serviceUpdate) (*configSelector, erro
 	}
 
 	for i, rt := range su.virtualHost.Routes {
+		clusters := newWRR()
 		if rt.ClusterSpecifierPlugin != "" {
+			clusters.Add(&routeCluster{
+				name:                     "cluster:" + rt.ClusterSpecifierPlugin,
+			}, 1)
+
 			ci := r.activeClusters["cluster_specifier_plugin:" + rt.ClusterSpecifierPlugin]
 			if ci == nil {
 				ci = &clusterInfo{refCount: 0}
 				r.activeClusters["cluster_specifier_plugin:" + rt.ClusterSpecifierPlugin] = ci
 			}
 			cs.clusters["cluster_specifier_plugin:" + rt.ClusterSpecifierPlugin] = ci
```

Comment:
Can this bit be shared with the below section?  Maybe factored into another function?